### DISCUSSION TITLE
SIP-4 (package.json) Discussion

### DISCUSSION THREAD

**Discussion by ritave** [MetaMask/SIPs#51] [SIPs]
This thread is meant to be a general place for open discussion concerning the state of [SIP-4 (package.json)](https://github.com/MetaMask/SIPs/blob/main/SIPS/sip-4.md).

The SIP proposes merging of the manifest and package.json to have one package file with definitions.

**Comment by BeroBurny** [comment]
It does not like the idea of making it as a SIP, and even less like the idea of deprecating `snap.manifest.json`.
Another way it is okay to have an option is to choose a preferred way of doing it.

**Reply by ritave** [reply]
You wouldn't be able to comment on it if it weren't a SIP. This the whole point of them - to get the community involved.

Can you expand why you don't like deprecating `snap.manifest.json`? Developers complain that the data desyncs between `package.json` and the manifest.

**Comment by tmpfs** [comment]
I have been working on a [package registry backed by IPFS](https://github.com/LavaMoat/ipfs-registry) and the API naturally exposes all the data in `package.json` so having the snaps manifest data in `package.json` would make it easier to retrieve the manifest programmatically without needing to extract files from the tarball.

So my vote is to deprecate `snap.manifest.json` 👍

**Reply by ritave** [reply]
This is a _cool_ project!

**Comment by eriknson** [comment]
Less files—less confusions

**Comment by ritave** [comment]
The MetaMask team decided to stay with `snap.manifest.json` for the for now, and so this SIP is being moved to Withdrawn.

Should we want to revisit this decision in the future, a new SIP will be created.
